lib/database.js

```diff
@@ -1,40 +1,40 @@
 const path = require('path');
 
+const Sequelize = require('sequelize');
 const { Client } = require('pg');
 const Umzug = require('umzug');
 
-const SequelizeDb = require('../models');
-
 const env = process.env.NODE_ENV || 'development';
 const DEFAULT_CONFIG = require('../config/config')[env];
 
 class Database {
     constructor(config) {
         this.config = config || DEFAULT_CONFIG;
-        this.SequelizeDb = SequelizeDb;
+        this.SequelizeDb = Database.createSeqelize(config);
+    }
+
+    static createSeqelize(config = DEFAULT_CONFIG) {
+        if (config.use_env_variable) {
+            return new Sequelize(process.env[config.use_env_variable]);
+        }
+        return new Sequelize(config.database, config.username, config.password, config);
     }
 
     isAvailable() {
-        const { username, password, host } = this.config;
-        this.client = new Client({
-            user: username,
-            password,
-            host,
-            database: 'postgres',
-        });
-        return new Promise((resolve) => {
-            this.client.connect((e) => {
-                this.client.end();
-                if (e) {
-                    resolve(false);
-                } else {
-                    resolve(true);
-                }
-            });
-        });
+        if (this.isSqlite()) {
+            return Promise.resolve(true);
+        }
+
+        return this.SequelizeDb.authenticate()
+            .then(() => true)
+            .catch(() => false);
     }
 
     createDatabase(dbName) {
+        if (this.isSqlite()) {
```

Comment:
Is there no concept of DB creation in SQlite? Surprised there's not an equivalent set of steps here (eg. write the file with the schema).

Reply:
not really- once you open a sqlite db if the file doesn't exist it just creates it. migrations/schema are still controlled by sequelize